unittests/test_cli.py

```diff
@@ -301,22 +301,37 @@ def test_check_sanity_failure(run_reframe, tmp_path):
 def test_check_sanity_failure_dry_run_mode(run_reframe, tmp_path):
     returncode, stdout, stderr = run_reframe(
         checkpath=['unittests/resources/checks/frontend_checks.py'],
-        more_options=['-n', 'SanityFailureCheck', '-S', 'dry_run_mode=1']
+        more_options=['-n', 'SanityFailureCheck', '--dry-run']
     )
     assert 'PASSED' in stdout
-
-    # This is a normal failure, it should not raise any exception
     assert 'Traceback' not in stdout
     assert 'Traceback' not in stderr
     assert returncode == 0
     assert os.path.exists(
-        tmp_path / 'output' / 'generic' / 'default' /
+        tmp_path / 'stage' / 'generic' / 'default' /
         'builtin' / 'SanityFailureCheck' / 'rfm_job.sh'
     )
     assert not os.path.exists(
-        tmp_path / 'output' / 'generic' / 'default' /
+        tmp_path / 'stage' / 'generic' / 'default' /
         'builtin' / 'SanityFailureCheck' / 'rfm_job.out'
     )
+    assert not os.path.exists(
+        tmp_path / 'output' / 'generic' / 'default' /
+        'builtin' / 'SanityFailureCheck' / 'rfm_job.sh'
+    )
+
+
+def test_check_sanity_failure_distribute_dry_run(run_reframe, tmp_path):
```

Comment:
Instead of copy-pasting the tests, I suggest to have parameterised fixture to effectively duplicate them. Something like this:

```python
@pytest.fixture(params=['run', 'dry_run'])
def action(request):
    return request.param

def test_check_sanity_failure_distribute(run_reframe, tmp_path, action):
    ...
```

We will also need a test to check the `--dry-run` and  `--repeat` combination.